src/ol/style/IconImage.js

```diff
@@ -125,10 +121,11 @@ class IconImage extends EventTarget {
     if (this.size_) {
       this.image_.width = this.size_[0];
       this.image_.height = this.size_[1];
+    } else {
+      this.size_ = [this.image_.width, this.image_.height];
     }
-    this.size_ = [this.image_.width, this.image_.height];
     this.unlistenImage_();
-    this.replaceColor_();
+    this.replaceColor_(1);
```

Comment:
I think this call is not needed as the call is done with the correct pixel ratio when the image is rendered.

Reply:
I agree, I will remove it :-)